### PR TITLE
Register Interpolations for Symbolics

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
@yingboma do you know why the printing is weird?

```julia
julia> using Symbolics; u = 2.0collect(1:10); t = 1.0collect(1:10);

julia> A = LinearInterpolation(u,t);

julia> @variables t
1-element Vector{Num}:
 t

julia> A(t)
[2.0, 4.0, 6.0, 8.0, 10.0, 12.0, 14.0, 16.0, 18.0, 20.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0](t)

julia> Symbolics.operation(A(t))
20-element LinearInterpolation{Vector{Float64}, Vector{Float64}, true, Float64}:
  2.0
  4.0
  6.0
  8.0
 10.0
  ⋮
  6.0
  7.0
  8.0
  9.0
 10.0

julia> Symbolics.arguments(A(t))
1-element Vector{Num}:
 t
```

### CODE REVIEW

File: src/DataInterpolations.jl [hunk 1]
```diff
@@ -31,6 +31,13 @@ ChainRulesCore.frule((_, _, Δt), ::typeof(_interpolate), A::AbstractInterpolati
 
 (interp::AbstractInterpolation)(t::Number) = _interpolate(interp, t)
 
+import Symbolics
+Base.@pure __parameterless_type(T) = Base.typename(T).wrapper
```
Comment (andreasnoack) [inline src/DataInterpolations.jl:35]: What is this function used for?

Comment (ChrisRackauckas) [reply src/DataInterpolations.jl:35]: `Array{Float64} -> Array` for the op name.

Comment (andreasnoack) [reply src/DataInterpolations.jl:35]: But where is it used?

Comment (ChrisRackauckas) [reply src/DataInterpolations.jl:35]: good point.

File: src/DataInterpolations.jl [hunk 2]
```diff
@@ -31,6 +31,13 @@ ChainRulesCore.frule((_, _, Δt), ::typeof(_interpolate), A::AbstractInterpolati
...
+Base.nameof(x::AbstractInterpolation) = typeof(x)
```
Comment (YingboMa) [inline src/DataInterpolations.jl:36]: `nameof` should return `Symbol`

File: src/DataInterpolations.jl [hunk 3]
```diff
@@ -31,6 +31,13 @@ ChainRulesCore.frule((_, _, Δt), ::typeof(_interpolate), A::AbstractInterpolati
...
+Base.isbinaryoperator(::Type{T}) where T <: AbstractInterpolation = false
```
Comment (YingboMa) [inline src/DataInterpolations.jl:37]: This is should be removed.

File: src/DataInterpolations.jl [hunk 4]
```diff
@@ -31,6 +31,13 @@ ChainRulesCore.frule((_, _, Δt), ::typeof(_interpolate), A::AbstractInterpolati
...
+(interp::DataInterpolations.AbstractInterpolation)(t::Symbolics.Num) = Symbolics.SymbolicUtils.term(interp,t)
```
Comment (YingboMa) [inline src/DataInterpolations.jl:38]: Needs `unwrap` on `t`.

Comment (ChrisRackauckas) [reply src/DataInterpolations.jl:38]: Yeah these were all handled in https://github.com/PumasAI/DataInterpolations.jl/pull/94

**Pull request merged**